File: shared_resources/src/main/java/com/thewizrd/shared_resources/helpers/ValueAction.java
package com.thewizrd.shared_resources.helpers;

public final class ValueAction extends Action {
    private ValueDirection direction;

    private ValueAction() {
        super();
    }

    public ValueAction(Actions action, ValueDirection direction) {
        super(action);
        this.direction = direction;
    }

    public ValueDirection getDirection() {
        return direction;
    }

    public void setDirection(ValueDirection direction) {
        this.direction = direction;
    }
}

File: shared_resources/src/main/java/com/thewizrd/shared_resources/actions/ValueAction.kt
package com.thewizrd.shared_resources.actions

open class ValueAction(action: Actions, var direction: ValueDirection) : Action(action) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is ValueAction) return false

        if (direction != other.direction) return false

        return true
    }

    override fun hashCode(): Int {
        return direction.hashCode()
    }
}